babel.dependencies.config.js

```diff
@@ -0,0 +1,31 @@
+/** @format */
+const config = {
+	// see https://github.com/webpack/webpack/issues/4039#issuecomment-419284940
+	sourceType: 'unambiguous',
+	presets: [
+		[
+			'@babel/env',
+			{
+				modules: false,
+				targets: { browsers: [ 'last 2 versions', 'Safari >= 10', 'iOS >= 10', 'ie >= 11' ] },
+				useBuiltIns: 'entry',
+				shippedProposals: true, // allows es7 features like Promise.prototype.finally
+			},
+		],
+		'@babel/react',
+	],
+	plugins: [
+		'@babel/plugin-syntax-dynamic-import',
```

Comment:
This transform is also not needed. We don't expect `import( ... ).then( ... )` statements in NPM packages, do we?

Reply:
I don't think so, but CRA has it.... It would be odd.

Reply:
To CRA it was added by @Timer in https://github.com/facebook/create-react-app/pull/5047. The PR description doesn't mention that any published NPM package would use it.

It's a very minor issue that doesn't really matter at all. It's just my curiosity that's not 100% satisfied 😃

Reply:
> This transform is also not needed. We don't expect import( ... ).then( ... ) statements in NPM packages, do we?

We expect npm packages to use it in Create React App.

Reply:
Also, compiling `node_modules` is not without its pitfalls, it's very unsafe.
You should avoid compiling `node_modules if you can. 😄

Reply:
> You should avoid compiling `node_modules` if you can. 😄

Given the direction that many dependencies are going, dropping support for Node 4, I don't really think that's viable. At least not for IE11 and other downlevel browsers.

I'm not entirely sure what's unsafe about it though. @Timer can you share anything that CRA has run into?

Reply:
Some code just isn't safe to compile, or you hit edge cases when you double minify.
Mapbox GL was a big one that was broken, but many things that use web workers break IIRC.

The reasons for breaks are so nebulous I can't really quantify them down to a handful of reasons, sorry!

Reply:
@Automattic/calypso-sdk @jeffersonrabb could you give this PR a spin for the map block and see if we have problems?

Reply:
Hi @blowery - I tried the PR out for Map and something is definitely broken. I compared this branch to `master` - on `master` all is well, but on this one the map tiles don't seem to be loading. Lots of errors in the console, although they're not exactly clear:

```blob:http://localhost/9821f18e-55d0-4fea-a0cb-16ec87408991:1 Uncaught ReferenceError: _typeof is not defined
    at tr (blob:http://localhost/9821f18e-55d0-4fea-a0cb-16ec87408991:1)
    at dr (blob:http://localhost/9821f18e-55d0-4fea-a0cb-16ec87408991:1)
    at new bn (blob:http://localhost/9821f18e-55d0-4fea-a0cb-16ec87408991:1)
    at new Cn (blob:http://localhost/9821f18e-55d0-4fea-a0cb-16ec87408991:1)
    at blob:http://localhost/9821f18e-55d0-4fea-a0cb-16ec87408991:1
    at blob:http://localhost/9821f18e-55d0-4fea-a0cb-16ec87408991:1```

Reply:
thanks @jeffersonrabb! i'll add mapbox to the blacklist.

Reply:
added in https://github.com/Automattic/wp-calypso/pull/30196/commits/34a434b381b17a4f66b1439b429e22a84e9cf262

Reply:
though that does make me revisit the choice to use a blacklist. Maybe we need a whitelist instead.

Reply:
I'd suggest keeping a list of "bad actors". If packages are intended to be used in the browser (even if they're Node packages) the publisher is responsible for compiling.
We always told users who tried to import non-ES5 that the publisher had to fix, and most of the time, they did.

Bad actors would just contain often-used packages that refuse to compile themselves.

Reply:
Insider tip: be sure to disable `transform-typeof-symbol`. This should fix mapbox and reduce some of the side effects of compiling `node_modules` if you go through with it.

Reply:
switch all this over to using a whitelist